README
# clj-assorted-utils

Unsorted bunch of helper and utility functions.

## Usage

Have a look at the tests and doc strings.

## License

Copyright (C) 2012 Ruediger Gad

Distributed under the Eclipse Public License, the same as Clojure.



changelog.txt
master

1.4.0
  - Add println-err. Mark print-err-ln as deprecated.
    This should be more idiomatic with respect to naming conventions.

1.3.1
  - Fix shutdown hook functionality.
  - Fix shutdown hook functionality.

1.3.0
  - Add convenience function for adding shutdown hooks.
1.2.4
  - Increment version number to make a clean release.

1.2.3
  - Assure that output threads terminate.

1.2.2
  - Do not kill the process in exit-with-out when there are no more lines
  available.
  - Refactor exec-with-out such that it is also possible to read from stderr.

1.2.1
  - Return process from exit-with-out.

1.2.0
  - Add more powerful and convenient to use counter.
  - Add convenience function for processing input line-by-line.

1.1.0
  - Add functions for serializing and compressing objects to byte arrays.

1.0.0
  - First release under new name clj-assorted-utils.


test/clj_assorted_utils/test/util.clj
;;;
;;;   Copyright 2012, 2013, Ruediger Gad
;;;
;;;   This software is released under the terms of the Eclipse Public License 
;;;   (EPL) 1.0. You can find a copy of the EPL at: 
;;;   http://opensource.org/licenses/eclipse-1.0.php
;;;

(ns
  ^{:author "Ruediger Gad",
    :doc "Unit tests for utility and helper functions"}
  clj-assorted-utils.test.util
  (:use clojure.test
        clojure.test.junit
        clojure.java.io
        clj-assorted-utils.util))


;(defn junit-output-fixture [f]
;  (with-open [wrtr (writer (str *ns* ".xml"))]
;    (binding [*test-out* wrtr]
;      (with-junit-output
;        (f)))))
;(use-fixtures :once junit-output-fixture)


;;;
;;; Tests for executing commands.
;;;
(deftest test-exec-with-out
  (let [command "ls /etc/passwd"
        stdout-run (prepare-flag)
        stdout-fn (fn [_] (set-flag stdout-run) nil)]
    (exec-with-out command stdout-fn)
    (sleep 100)
    (is (flag-set? stdout-run))))
    
(deftest test-exec-with-out-process-output
  (let [command "ls /etc/passwd"
        stdout-run (prepare-flag)
        stdout-fn (fn [out] (if (= out "/etc/passwd") (set-flag stdout-run)))]
    (exec-with-out command stdout-fn)
    (sleep 100)
    (is (flag-set? stdout-run))))



;;;
;;; Tests for handling files and directories.
;;;
(def test-dirname "test-dir-foo")
(def test-filename "test-file-bar")

(deftest test-exists
  (is (exists? "test/clj_assorted_utils/test/util.clj"))
  (is (exists? "test")))

(deftest test-is-file
  (is (is-file? "test/clj_assorted_utils/test/util.clj"))
  (is (not (is-file? "test"))))

(deftest test-file-exists
  (is (file-exists? "test/clj_assorted_utils/test/util.clj"))
  (is (not (file-exists? "test")))
  (is (not (file-exists? test-filename))))

(deftest test-is-dir
  (is (is-dir? "test"))
  (is (not (is-dir? "test/clj_assorted_utils/test/util.clj"))))

(deftest test-dir-exists
  (is (dir-exists? "test"))
  (is (not (dir-exists? "test/clj_assorted_utils/test/util.clj")))
  (is (not (dir-exists? test-dirname))))

(deftest mkdir-rmdir
  (let [dirname test-dirname]
    (is (not (dir-exists? dirname)))
    (mkdir dirname)
    (is (dir-exists? dirname))
    (rmdir dirname)
    (is (not (dir-exists? dirname)))))

(deftest touch-rm
  (let [filename test-filename]
    (is (not (file-exists? filename)))
    (touch filename)
    (is (file-exists? filename))
    (rm filename)
    (is (not (file-exists? filename)))))



;;;
;;; Tests for setting and querying flags and a simple counter.
;;;
(deftest flag-not-set
  (let [flag (prepare-flag)]
    (is (not (flag-set? flag)))))

(deftest flag-set
  (let [flag (prepare-flag)]
    (set-flag flag)
    (is (flag-set? flag))))

(deftest counter-test
  (let [my-counter (prepare-counter)]
    (dotimes [_ 1000] (inc-counter my-counter))
    (is (= 1000 @my-counter))))

(deftest counter-with-initial-value-test
  (let [my-counter (prepare-counter 1000)]
    (dotimes [_ 1000] (inc-counter my-counter))
    (is (= 2000 @my-counter))))

(deftest counter-convenience-test
  (let [my-counter (counter)]
    (dotimes [_ 1000] (my-counter inc))
    (is (= 1000 (my-counter)))))

(deftest counter-convenience-test-custom-fn
  (let [my-counter (counter)]
    (dotimes [_ 1000] (my-counter #(+ 2 %)))
    (is (= 2000 (my-counter)))))

(deftest counter-convenience-test-init-value-decrementing
  (let [my-counter (counter 1000)]
    (dotimes [_ 1000] (my-counter dec))
    (is (= 0 (my-counter)))))



;;;
;;; Tests for getting class and fn names.
;;;
(deftest get-classname
  (let [o (Object.)
        n (classname o)]
    (is (= "Object" n))))

(defn test-fn [] (println "It's -O3 the letter not -03 the number."))

(deftest test-fn-name
  (is (= "test-fn" (fn-name test-fn))))


;;;
;;; Tests for manipulating vectors.
;;;
(deftest test-byte-seq-to-int
  (let [byte-vec [82 17 0 0]
        int-val 4434]
    (is (= int-val (byte-seq-to-int byte-vec)))))

(deftest test-get-int-from-byte-vector
  (let [byte-vec [121 -110 84 79 0 0 0 0 -23 -71 8 0 0 0 0 0 82 17 0 0 82 17 0 0]
        int-val 4434]
    (is (= int-val (get-int-from-byte-vector byte-vec 16))))) 

(deftest test-int-to-byte-vector
  (let [int-val 4434
        byte-vec [82 17 0 0]]
    (is (= byte-vec (subvec (int-to-byte-vector int-val) 0 4)))))

(deftest test-vec-replace
  (let [original-vec [1 2 3 4 5 6]
        expected-vec [1 2 "a" "b" "c" 6]
        changed-vec (vec-replace original-vec 2 ["a" "b" "c"])]
    (is (= expected-vec changed-vec))))

(deftest test-change-int-in-byte-vector
  (let [original-vec [121 -110 84 79 0 0 0 0 -23 -71 8 0 0 0 0 0 82 17 0 0 82 17 0 0]
        expected-vec [121 -110 84 79 0 0 0 0 -23 -71 8 0 0 0 0 0 70 17 0 0 82 17 0 0]
        changed-vec (change-int-in-byte-vector original-vec 16 #(- % 12))]
    (is (= expected-vec changed-vec))))



;;;
;;; Tests for messing with XML.
;;; Primarily for transforming XML data in string format to maps.
;;;
(deftest test-xml-string-to-map
  (let [xml-str "<foo fubar=\"snafu\">bar</foo>"
        expected-map {:tag :foo :attrs {:fubar "snafu"} :content ["bar"]}]
    (is (= expected-map (xml-string-to-map xml-str)))))

(deftest test-stringify-keyword
  (is (= "foo" (stringify-keyword :foo))))

(deftest test-stringify-map
  (let [input-map {:tag :foo :attrs {:fubar "snafu"} :content ["bar"]}
        expected-map {"tag" "foo" "attrs" {"fubar" "snafu"} "content" ["bar"]}]
    (is (= expected-map (stringify-map input-map)))))

(deftest test-xml-string-to-map-stringified
  (let [xml-str "<foo fubar=\"snafu\">bar</foo>"
        expected-map {"tag" "foo" "attrs" {"fubar" "snafu"} "content" ["bar"]}]
    (is (= expected-map (xml-string-to-map-stringified xml-str)))))



;;;
;;; Tests for running fn's regularly
;;;
(deftest test-simple-executor
  (let [flag (prepare-flag)
        run-fn #(set-flag flag)
        exec (executor)]
    (run-once exec run-fn 100)
    (sleep 300)
    (is (flag-set? flag))))



;;;
;;; Tests for object serialization.
;;;
(deftest test-object-to-byte-array
  (let [obj-int (int 123)
        obj-string "abc"
        ba-int (object-to-byte-array obj-int)
        ba-string (object-to-byte-array obj-string)]
    (is (not (nil? ba-int)))
    (is (not (nil? ba-string)))))

(deftest test-compress-object-to-byte-array
  (let [obj-int (int 123)
        obj-string "abc"
        ba-int (compress-object-to-byte-array obj-int :gzip)
        ba-string (compress-object-to-byte-array obj-string :gzip)]
    (is (not (nil? ba-int)))
    (is (not (nil? ba-string)))))



;;;
;;; Tests for reading input line-by-line
;;;
(deftest process-line-by-line-test
  (let [cntr (counter)]
    (is (= 0 (cntr)))
    (process-line-by-line "test-file.txt" (fn [_] (cntr inc)))
    (is (= 10 (cntr)))))

(deftest process-line-by-line-test2
  (let [cntr (counter)]
    (is (= 0 (cntr)))
    (process-line-by-line "test-file.txt" (fn [x] (cntr #(+ % (read-string x)))))
    (is (= 55 (cntr)))))



autodoc/index.clj
{:namespaces
 ({:source-url nil,
   :wiki-url "clj-assorted-utils.util-api.html",
   :name "clj-assorted-utils.util",
   :author "Ruediger Gad",
   :doc "Utility and helper functions"}),
 :vars
 ({:arglists ([byte-seq]),
   :name "byte-seq-to-int",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/byte-seq-to-int",
   :doc "Convert the byte sequence byte-seq to an integer.",
   :var-type "function",
   :line 251,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([v n f]),
   :name "change-int-in-byte-vector",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/change-int-in-byte-vector",
   :doc
   "In the given vector v, change the byte representation of an integer by applying function f.",
   :var-type "function",
   :line 294,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([o]),
   :name "classname",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/classname",
   :doc
   "Get classname without leading package names of the given object o.",
   :var-type "function",
   :line 230,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([obj] [obj alg]),
   :name "compress-object-to-byte-array",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/compress-object-to-byte-array",
   :doc
   "Write an object into a byte array and compress it.\nA newly allocated byte array with the serialized and compressed object is returned.\nThe compression algorithm can be specified via the second parameter.\nCurrently GZIP (:gzip) and ZIP (:zip) are supported.\nThe default is GZIP.",
   :var-type "function",
   :line 371,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([] [init]),
   :name "counter",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/counter",
   :doc
   "Convenience function for creating a more powerful counter.\nThe created counter allows to have arbitrary single-argument functions passed for manipulating the internal value.\nWhen the created counter is called with no argument the current value is returned.\nExample:\n(use 'clj-assorted-utils.util)\n(let [cntr (counter 123)]\n  (println (cntr))\n  (cntr inc)\n  (cntr dec)\n  (cntr #(- % 123))\n  (cntr))",
   :var-type "function",
   :line 203,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([d & body]),
   :name "delay-eval",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/delay-eval",
   :doc "Evaluates the supplied body with the given delay 'd' ([ms]).",
   :var-type "macro",
   :line 125,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([d]),
   :name "dir-exists?",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/dir-exists?",
   :doc "Returns true if f exists and is a directory.",
   :var-type "function",
   :line 93,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([cmd]),
   :name "exec",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/exec",
   :doc "Execute the given command.",
   :var-type "function",
   :line 27,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([cmd]),
   :name "exec-blocking",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/exec-blocking",
   :doc
   "Execute the given command and block until execution has finished.",
   :var-type "function",
   :line 32,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([cmd stdout-fn]),
   :name "exec-with-out",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/exec-with-out",
   :doc
   "Execute the given command and process the output written to stdout with stdout-fn.\nstdout-fn is run in an own thread.\nThe data written to stdout is passed to stdout-fn line-by-line.\nWhen a value of nil is read processing stops and the process in which cmd was executed is destroyed.",
   :var-type "function",
   :line 37,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([]),
   :name "executor",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/executor",
   :doc "Create an executor for executing fns in an own thread.",
   :var-type "function",
   :line 134,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "exists?",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/exists?",
   :doc "Returns true if f exists in the filesystem.",
   :var-type "function",
   :line 71,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "file-exists?",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/file-exists?",
   :doc "Returns true if f exists and is a file.",
   :var-type "function",
   :line 81,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "flag-set?",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/flag-set?",
   :doc
   "Test if flag had been set.\nReturns true if flag was set and false otherwise.",
   :var-type "function",
   :line 187,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "fn-name",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/fn-name",
   :doc "Get the name of the given fn f.",
   :var-type "function",
   :line 236,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([v n]),
   :name "get-int-from-byte-vector",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/get-int-from-byte-vector",
   :doc
   "From given vector of bytes v get the integer starting at offset n.\nStarting at offset n the next 4 bytes will be converted to an integer.",
   :var-type "function",
   :line 269,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([c]),
   :name "inc-counter",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/inc-counter",
   :doc "Increment the given simple counter c.",
   :var-type "function",
   :line 198,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([val]),
   :name "int-to-byte-vector",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/int-to-byte-vector",
   :doc "Convert the given integer val to a vector of 4 bytes.",
   :var-type "function",
   :line 276,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([d]),
   :name "is-dir?",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/is-dir?",
   :doc "Returns true if f is a directory.",
   :var-type "function",
   :line 88,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "is-file?",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/is-file?",
   :doc "Returns true if f is a file.",
   :var-type "function",
   :line 76,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([d]),
   :name "mkdir",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/mkdir",
   :doc
   "Create directory d and if required all parent directories.\nThis is equivalent to 'mkdir -p d' on Linux systems.",
   :var-type "function",
   :line 100,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([obj]),
   :name "object-to-byte-array",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/object-to-byte-array",
   :doc
   "Write the given object into a byte array.\nA newly allocated byte arry with the serialized object is returned.",
   :var-type "function",
   :line 362,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([] [init]),
   :name "prepare-counter",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/prepare-counter",
   :doc "Prepare a simple counter. Use @ to access the value.",
   :var-type "function",
   :line 193,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([]),
   :name "prepare-flag",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/prepare-flag",
   :doc "Prepare a flag with default value false.",
   :var-type "function",
   :line 176,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([& s]),
   :name "print-err",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/print-err",
   :doc "print to stderr.",
   :var-type "function",
   :line 345,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([& s]),
   :name "print-err-ln",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/print-err-ln",
   :doc "println to stderr.",
   :var-type "function",
   :line 351,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([location f]),
   :name "process-line-by-line",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/process-line-by-line",
   :doc
   "Process input from location line-by-line.\nEach line is passed to f.\nThe processing is done with a clojure reader.\nSo all locations supported by reader are automatically supported as well.",
   :var-type "function",
   :line 393,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "rm",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/rm",
   :doc "Delete file f.",
   :var-type "function",
   :line 106,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([d]),
   :name "rmdir",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/rmdir",
   :doc "Delete d if d is an empty directory.",
   :var-type "function",
   :line 111,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([exec f d] [exec f d tu]),
   :name "run-once",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/run-once",
   :doc
   "Run f using executor exec once with delay d.\nOptionally a time unit tu can be given.\ntu defaults to TimeUnit/MILLISECONDS.",
   :var-type "function",
   :line 149,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([exec f d] [exec f id d] [exec f id d tu]),
   :name "run-repeat",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/run-repeat",
   :doc
   "Run f repeatedly using executor exec with delay d.\nOptionally an initial delay id and a time unit tu can be given.\nTime unit is a static member of TimeUnit, e.g., TimeUnit/SECONDS \nand defaults to TimeUnit/MILLISECONDS.",
   :var-type "function",
   :line 158,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "set-flag",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/set-flag",
   :doc "Set flag to true.",
   :var-type "function",
   :line 181,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([exec]),
   :name "shutdown",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/shutdown",
   :doc "Shut executor down.",
   :var-type "function",
   :line 139,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([exec]),
   :name "shutdown-now",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/shutdown-now",
   :doc "Force executor shut down.",
   :var-type "function",
   :line 144,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([ms]),
   :name "sleep",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/sleep",
   :doc "Sleep for the given number of milliseconds.",
   :var-type "function",
   :line 22,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([k]),
   :name "stringify-keyword",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/stringify-keyword",
   :doc
   "If a keyword is passed returns the name of the keyword.\nElse the input is left unchanged.",
   :var-type "function",
   :line 314,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([m]),
   :name "stringify-map",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/stringify-map",
   :doc
   "Convert _all_ keywords in a map to their respective names.\nThis, essentially, is an extended version of clojure.walk/stringify-keys,\nwhich only converts the keys to strings.",
   :var-type "function",
   :line 322,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([f]),
   :name "touch",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/touch",
   :doc "If f does not exist, create it.",
   :var-type "function",
   :line 116,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([v n delta]),
   :name "vec-replace",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/vec-replace",
   :doc
   "In given vector v replace the content of v starting at index n with delta.",
   :var-type "function",
   :line 286,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([xml-str]),
   :name "xml-string-to-map",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/xml-string-to-map",
   :doc
   "Takes an XML definition in form of a string and outputs the corresponding map.",
   :var-type "function",
   :line 307,
   :file "src/clj_assorted_utils/util.clj"}
  {:arglists ([xml-str]),
   :name "xml-string-to-map-stringified",
   :namespace "clj-assorted-utils.util",
   :source-url nil,
   :raw-source-url nil,
   :wiki-url
   "/clj-assorted-utils.util-api.html#clj-assorted-utils.util/xml-string-to-map-stringified",
   :doc
   "Convert the XML string xml-str to a map with all keywords converted to strings.",
   :var-type "function",
   :line 334,
   :file "src/clj_assorted_utils/util.clj"})}
